cmd/metrics.go

```diff
@@ -81,121 +85,195 @@ func (c *minioCollector) Collect(ch chan<- prometheus.Metric) {
 	// Expose MinIO's version information
 	minioVersionInfo.WithLabelValues(Version, CommitID).Add(1)
 
-	// Always expose network stats
+	// Fetch disk space info
+	objLayer := newObjectLayerFn()
+	// Service not initialized yet
+	if objLayer == nil {
+		return
+	}
 
-	// Network Sent/Received Bytes
-	ch <- prometheus.MustNewConstMetric(
-		prometheus.NewDesc(
-			prometheus.BuildFQName("minio", "network", "sent_bytes_total"),
-			"Total number of bytes sent by current MinIO server instance",
-			nil, nil),
-		prometheus.CounterValue,
-		float64(globalConnStats.getTotalOutputBytes()),
-	)
-	ch <- prometheus.MustNewConstMetric(
-		prometheus.NewDesc(
-			prometheus.BuildFQName("minio", "network", "received_bytes_total"),
-			"Total number of bytes received by current MinIO server instance",
-			nil, nil),
-		prometheus.CounterValue,
-		float64(globalConnStats.getTotalInputBytes()),
-	)
+	for _, endpoint := range globalEndpoints {
```

Comment:
I think we should just use getDisksInfo() here or improve StorageInfo() call to provide this info per disks as well.

Reply:
As these metrics are labelled to endpoints. getDisksInfo() will not be useful for endpoint specific disk info query. And StorageInfo() change for this will be of bigger scope. Will parallelize the calls for better performance :+1: